source/core/simulation_control.cc

```diff
@@ -177,6 +177,37 @@ SimulationControl::read(std::string prefix)
   time_step = time_step_vector[0];
 }
 
+std::vector<double>
+SimulationControl::get_checkpointed_simulation_control_info(std::string prefix)
+{
+  std::string   filename = prefix + ".simulationcontrol";
+  std::ifstream input(filename.c_str());
+  AssertThrow(input, ExcFileNotOpen(filename));
+
+  // Store the time steps and last checkpointed time without modifying the
+  // simulation control information
+  std::vector<double> time_steps(time_step_vector.size());
+  double              last_checkpointed_time;
+
+  // Returned vector with the last checkpointed time and time step
+  std::vector<double> simulation_control_info(2);
+
+  std::string buffer;
+  std::getline(input, buffer);
+  for (unsigned int i = 0; i < time_step_vector.size(); ++i)
+    input >> buffer >> time_steps[i];
+  input >> buffer >> buffer;
+  input >> buffer >> last_checkpointed_time;
+  input >> buffer >> buffer;
+
+  // Fix the current time as final checkpointed time
+  simulation_control_info[0] = last_checkpointed_time;
+
+  // Fix time step to be the last time_step that was used
+  simulation_control_info[1] = time_steps[0];
```

Comment:
You really want to have the new simulation start at the time the previous one finished? why not restart it from zero? does that really pose a problem? I'm not sure what's the goal her (to me this is not intuitive, because if you set the initial condition, shouldnt the time be at zero still ? )

Reply:
This is only to get end time for the average velocity calculation! The simulation time will be restarted from zero in the second simulation. This is necessary because of the way we calculate the average velocity.

Reply:
:)perfect